#! /bin/sh /usr/share/dpatch/dpatch-run
## 40_fix_spurious_dollar_signs_added_to_command_lines.dpatch by Alexander Wirt <formorer@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes the output of spurious $ signs in commandoutput (#480001)

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' pkg-nagios3~/common/macros.c pkg-nagios3/common/macros.c
--- pkg-nagios3~/common/macros.c	2012-05-07 20:43:10.000000000 +0200
+++ pkg-nagios3/common/macros.c	2012-05-07 21:51:23.191049586 +0200
@@ -185,7 +185,8 @@
 				*output_buffer = (char *)realloc(*output_buffer, strlen(*output_buffer) + strlen(temp_buffer) + 3);
 				strcat(*output_buffer, "$");
 				strcat(*output_buffer, temp_buffer);
-				strcat(*output_buffer, "$");
+				if (buf_ptr!=NULL)
+					strcat(*output_buffer,"$");
 				}
 
 			/* insert macro */
